src/Native/System.Private.CoreLib.Native/pal_environment.cpp

```diff
@@ -30,3 +30,17 @@ extern "C" void CoreLibNative_Exit(int32_t exitCode)
 {
     exit(exitCode);
 }
+
+extern "C" char** CoreLibNative_EnvironGetSystemEnvironment()
+{
+    char** sysEnviron;
+
+#if HAVE__NSGETENVIRON
```

Comment:
We should also copy over the definition of `HAVE__NSGETENVIRON` in the config system to https://github.com/dotnet/corert/blob/master/src/Native/System.Private.CoreLib.Native/config.h.in and https://github.com/dotnet/corert/blob/master/src/Native/System.Private.CoreLib.Native/configure.cmake

Reply:
Having some odd trouble with this one. Maybe you have some thoughts. I added

```
#cmakedefine01 HAVE__NSGETENVIRON
```

to `config.h.in` and tried two different approaches for the definition. One is a copy of CoreCLR:
```
if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
  set(HAVE__NSGETENVIRON 1)
endif(CMAKE_SYSTEM_NAME STREQUAL Darwin)
```

Also tried 
```
check_function_exists(_NSGetEnviron HAVE__NSGETENVIRON)
```

Both scenarios set it (because I'm on macOS), however the build then fails with:
```
/Users/robertmiles3/Desktop/corert/src/Native/System.Private.CoreLib.Native/pal_environment.cpp:40:20: error: use of
      undeclared identifier '_NSGetEnviron'
    sysEnviron = *(_NSGetEnviron());
                   ^
```

Reply:
Try adding after the other includes at the top of the file:

```
#if HAVE__NSGETENVIRON
#include <crt_externs.h>
#endif
```

(CoreCLR version has separate `HAVE_CRT_EXTERNS_H` for this. It is not necessary.)